src/data_structures/qgram_index.rs

```diff
@@ -108,6 +108,11 @@ impl QGramIndex {
         &self.pos[self.address[qgram]..self.address[qgram + 1]]
     }
 
+    /// Iterate over the q-gram ranks of pattern
+    pub fn qgrams<'a>(&'a self, pattern: &'a [u8]) -> QGrams<std::slice::Iter<u8>> {
```

Comment:
This does not feel like the right place for such a method. If you don't want to recreate the RankTransform in your function above, maybe it is a better idea to make it a field of `Alphabet` (the overhead is quite limited). This way, everything can be easily accessed from `Alphabet`.